src/hw-monitor.cpp

```diff
@@ -140,66 +139,59 @@ namespace librealsense
     {
-        return _locked_transfer->send_receive(data);
+        return _locked_transfer->send_receive( data.data(), data.size() );
     }
 
     std::vector< uint8_t >
-    hw_monitor::send( command cmd, hwmon_response * p_response, bool locked_transfer ) const
+    hw_monitor::send( command const & cmd, hwmon_response * p_response, bool locked_transfer ) const
     {
-        hwmon_cmd newCommand(cmd);
-        auto opCodeXmit = static_cast<uint32_t>(newCommand.cmd);
+        uint32_t const opCodeXmit = cmd.cmd;
 
         hwmon_cmd_details details;
-        details.require_response = newCommand.require_response;
-        details.timeOut = newCommand.timeOut;
-
-        fill_usb_buffer(opCodeXmit,
-            newCommand.param1,
-            newCommand.param2,
-            newCommand.param3,
-            newCommand.param4,
-            newCommand.data,
-            newCommand.sizeOfSendCommandData,
-            details.sendCommandData.data(),
-            details.sizeOfSendCommandData);
+        details.require_response = cmd.require_response;
+        details.timeOut = cmd.timeout_ms;
+
+        fill_usb_buffer( opCodeXmit,
+                         cmd.param1,
+                         cmd.param2,
+                         cmd.param3,
+                         cmd.param4,
+                         cmd.data.data(),  // memcpy
+                         std::min( (uint16_t)cmd.data.size(), HW_MONITOR_BUFFER_SIZE ),
+                         details.sendCommandData.data(),
+                         details.sizeOfSendCommandData );
 
         if (locked_transfer)
         {
-            return _locked_transfer->send_receive({ details.sendCommandData.begin(),details.sendCommandData.end() });
+            return _locked_transfer->send_receive( details.sendCommandData.data(), details.sendCommandData.size() );
         }
 
         send_hw_monitor_command(details);
 
         // Error/exit conditions
         if (p_response)
             *p_response = hwm_Success;
-        if( !newCommand.require_response )
-            return std::vector<uint8_t>();
-
-        std::memcpy( newCommand.receivedOpcode, details.receivedOpcode.data(), 4 );
-        std::memcpy( newCommand.receivedCommandData,
-                     details.receivedCommandData.data(),
-                     details.receivedCommandDataLength );
-        newCommand.receivedCommandDataLength = details.receivedCommandDataLength;
+        if( ! cmd.require_response )
+            return {};
 
         // endian?
         auto opCodeAsUint32 = pack(details.receivedOpcode[3], details.receivedOpcode[2],
             details.receivedOpcode[1], details.receivedOpcode[0]);
         if (opCodeAsUint32 != opCodeXmit)
         {
             auto err_type = static_cast<hwmon_response>(opCodeAsUint32);
-            std::string err = hwmon_error_string(cmd, err_type);
-            LOG_DEBUG(err);
-            if (p_response)
+            //LOG_DEBUG(err);  // too intrusive; may be an expected error
```

Comment:
You removed the line that initialize `err`

`std::string err = hwmon_error_string(cmd, err_type);`

Better to keep as comment or remove both, right?

Reply:
OK, I'll remove